src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```diff
@@ -69,79 +100,164 @@ public void inferenceSentence(
+     * take any function or TaskType as currently we need to run only TextEmbedding tasks only.
+     *
+     * @param targetResponseFilters {@link List} of {@link String} which filters out the responses
+     * @param modelId               {@link String}
+     * @param inputText             {@link List} of {@link String} on which inference needs to happen
+     * @param listener              {@link ActionListener} which will be called when prediction is
+     *                              completed or errored out.
+     */
+    public void inferenceSentences(
+        @NonNull final List<String> targetResponseFilters,
+        @NonNull final String modelId,
+        @NonNull final List<String> inputText,
+        @NonNull final ActionListener<List<List<Float>>> listener
+    ) {
+        inferenceSentences(targetResponseFilters, modelId, inputText, null, listener);
+    }
+
+    /**
+     * Abstraction to call predict function of api of MLClient with provided targetResponse filters.
+     * It uses the custom model provided as modelId and run the {@link FunctionName#TEXT_EMBEDDING}.
+     * The return will be sent using the actionListener which will have a {@link List} of {@link List}
+     * of {@link Float} in the order of inputText. We are not making this function generic enough to
+     * take any function or TaskType as currently we need to run only TextEmbedding tasks only. Supports
+     * passing {@link MLAlgoParams} to the inference. If the model is asymmetric, passing a
+     * {@link org.opensearch.ml.common.input.parameter.textembedding.AsymmetricTextEmbeddingParameters}
+     * is mandatory. This method will check whether the model being used is asymmetric and correctly
+     * handle the parameter, so it's okay to always pass the parameter (even if the model is symmetric).
+     *
+     * @param targetResponseFilters {@link List} of {@link String} which filters out the responses
+     * @param modelId               {@link String}
+     * @param inputText             {@link List} of {@link String} on which inference needs to happen
+     * @param mlAlgoParams          {@link MLAlgoParams} which will be used to run the inference
+     * @param listener              {@link ActionListener} which will be called when prediction is
+     *                              completed or errored out.
      */
     public void inferenceSentences(
         @NonNull final List<String> targetResponseFilters,
         @NonNull final String modelId,
         @NonNull final List<String> inputText,
+        @Nullable final MLAlgoParams mlAlgoParams,
         @NonNull final ActionListener<List<List<Float>>> listener
     ) {
-        retryableInferenceSentencesWithVectorResult(targetResponseFilters, modelId, inputText, 0, listener);
+        retryableInferenceSentencesWithVectorResult(targetResponseFilters, modelId, inputText, mlAlgoParams, 0, listener);
```

Comment:
DTO object as argument size is exceeding 5 now.

Reply:
https://github.com/opensearch-project/neural-search/issues/790